### PR TITLE
Review ex1

### PR DESCRIPTION
**Pull request by Javiercerna**

### CODE REVIEW

File: exercise1/exc1/broadcast/scripts/broadcaster.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+import rospy
+from std_msgs.msg import Int16
+
+
+def talker():
+    pub = rospy.Publisher('xamin', Int16, queue_size=10)
+    rospy.init_node('broadcaster', anonymous=True)
```
Comment (Javiercerna) [inline exercise1/exc1/broadcast/scripts/broadcaster.py:8]:
```suggestion
    rospy.init_node('broadcaster', anonymous=True)
    pub = rospy.Publisher('xamin', Int16, queue_size=10)
```

I'm not sure if it makes a difference in this small example, but I would normally first initialize the node and then create the subscribers/publishers.

File: exercise1/exc1/broadcast/scripts/broadcaster.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    rate = rospy.Rate(20)  # 10hz
```
Comment (Javiercerna) [inline exercise1/exc1/broadcast/scripts/broadcaster.py:9]: Why did you write 10 Hz as a comment? Do you understand what `Rate(20)` means here? :slightly_smiling_face:

File: exercise1/exc1/broadcast/scripts/broadcaster.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        pub.publish(k)
+        rate.sleep()
+        k += 4
+
+
+if __name__ == '__main__':
+    try:
+        talker()
+    except rospy.ROSInterruptException:
+        pass
```
Comment (Javiercerna) [inline exercise1/exc1/broadcast/scripts/broadcaster.py:21]: Nice!

File: exercise1/exc1/receiver/scripts/receiver.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+import rospy
+from std_msgs.msg import Int16, Float32
+
+
+class Receiver:
+    pub = None
+    rate = None
```
Comment (Javiercerna) [inline exercise1/exc1/receiver/scripts/receiver.py:8]: This works well. But any reason why you didn't define this as part of a `__init__` method?

File: exercise1/exc1/receiver/scripts/receiver.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        rospy.init_node('receiver', anonymous=True)
+        rospy.Subscriber("xamin", Int16, self.callback)
+        self.pub = rospy.Publisher('/kthfs/result', Float32, queue_size=10)
+        self.rate = rospy.Rate(20)
+        rospy.spin()
+
+
+if __name__ == '__main__':
+    rec = Receiver()
+    rec.init()
```
Comment (Javiercerna) [inline exercise1/exc1/receiver/scripts/receiver.py:23]: Nice use of classes! Maybe to not confuse it with the built-in `__init__`, I would ask you to name this one `start`. But I'm being too picky :smile:

**Review by Javiercerna — COMMENTED** [review]
Good job! Not much to comment

File: exercise1/exc1/broadcast/scripts/broadcaster.py (hunk 2 above)
Comment (albertoxamin) [reply exercise1/exc1/broadcast/scripts/broadcaster.py:9]: Oops I forgot to change the comment from the ROS tutorial page 😅

**Pull request closed without merging**